src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -838,18 +837,30 @@ public static void TestingCreateInstanceFromObjectHandle(string physicalFileName
             Assert.True(File.Exists(physicalFileName));
         }
 
-        public static TheoryData<string, string, string, string, Type> TestingCreateInstanceFromObjectHandleData => new TheoryData<string, string, string, string, Type>
+        public static IEnumerable<object[]> TestingCreateInstanceFromObjectHandleData()
         {
+            Type exceptionType;
+
+            // Technically case-insensitive OS
+            if (PlatformDetection.IsNotMobile)
```

Comment:
I'd add a separate property to PlatformDetection instead of assuming that all mobile OSes are case sensitive. Something like https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.FileSystem/tests/FileSystemTest.cs#L14-L15